OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -436,57 +441,43 @@ public MovePart(Move move, WPos from, WPos to, WAngle fromFacing, WAngle toFacin
 
 			public override bool Tick(Actor self)
 			{
-				var ret = InnerTick(self, Move.mobile);
-
-				if (moveFraction > MoveFractionTotal)
-					moveFraction = MoveFractionTotal;
-
-				UpdateCenterLocation(self, Move.mobile);
-
-				if (ret == this)
-					return false;
+				var mobile = Move.mobile;
 
-				Queue(ret);
-				return true;
-			}
-
-			Activity InnerTick(Actor self, Mobile mobile)
-			{
-				moveFraction += mobile.MovementSpeedForCell(self, mobile.ToCell);
-				if (moveFraction <= MoveFractionTotal)
-					return this;
+				// Having non-zero progress in the first tick means that this MovePart is following on from
+				// a previous MovePart that has just completed during the same tick. In this case, we want to
+				// apply the carried over progress but not evaluate a full new step until the next tick.
+				if (!firstTick || progress == 0)
+					progress += mobile.MovementSpeedForCell(self, mobile.ToCell);
 
-				return OnComplete(self, mobile, Move);
-			}
+				firstTick = false;
 
-			void UpdateCenterLocation(Actor self, Mobile mobile)
-			{
-				// Avoid division through zero
-				if (MoveFractionTotal != 0)
+				if (progress >= Distance)
 				{
-					WPos pos;
-					if (EnableArc)
-					{
-						var angle = WAngle.Lerp(ArcFromAngle, ArcToAngle, moveFraction, MoveFractionTotal);
-						var length = int2.Lerp(ArcFromLength, ArcToLength, moveFraction, MoveFractionTotal);
-						var height = int2.Lerp(From.Z, To.Z, moveFraction, MoveFractionTotal);
-						pos = ArcCenter + new WVec(0, length, height).Rotate(WRot.FromYaw(angle));
-					}
-					else
-						pos = WPos.Lerp(From, To, moveFraction, MoveFractionTotal);
+					progress = Distance;
```

Comment:
This resets the carryover progress to 0, undoing the fix!

I can confirm that after removing this line unit movement does finally work as expected.